extension/src/router_channel.rs

```diff
@@ -450,9 +451,12 @@ impl RouterChannel {
       header_buffer.extend(b"\r\n");
       tx.send(Ok(Frame::data(header_buffer.into()))).await?;
 
-      let mut encoder: Option<GzEncoder<Vec<u8>>> = None;
+      let mut encoder: Option<GzEncoder<Writer<BytesMut>>> = None;
       if app_res_will_compress {
-        encoder = Some(GzEncoder::new(Vec::new(), flate2::Compression::default()));
+        encoder = Some(GzEncoder::new(
+          BytesMut::with_capacity(app_res_content_length as usize).writer(),
```

Comment:
This will most likely allocate a buffer larger than the compressed representation of the app response. Perhaps there is a better method to guess the size of the compressed data?